src/commands/init.command.ts

```diff
@@ -11,49 +12,70 @@ import { TopLevelCommand } from "./TopLevelCommand.ts";
 
 export function registerInitCommand(program: TopLevelCommand) {
   program
-    .command("init")
-    .description("initialize a new collie repository in the current directory")
-    .action(async (opts: GlobalCommandOptions) => {
-      const kit = new CollieRepository("./");
-      const logger = new Logger(kit, opts);
-
-      // ensure git is initialized
-      const cliFactory = new CliApiFacadeFactory(kit, logger);
-      const git = cliFactory.buildGit();
-
-      await git.init();
-
-      const dir = new DirectoryGenerator(WriteMode.skip, logger);
-      const d: Dir = {
-        name: "",
-        entries: [
-          { name: "README.md", content: readmeMd },
-          { name: ".gitignore", content: gitignore },
-          {
-            name: "kit",
-            entries: [
-              { name: ".gitkeep", content: "" },
-              { name: "README.md", content: kitReadmeMd },
-            ],
-          },
-          { name: "compliance", entries: [{ name: ".gitkeep", content: "" }] },
-          { name: "foundations", entries: [{ name: ".gitkeep", content: "" }] },
-        ],
-      };
-
-      await dir.write(d, "");
-
-      // this is the only place where an absolute path is ok, to show the user unambigously where
-      // the repository is on their file system
-      logger.progress(
-        "generated new collie repository at " + kit.resolvePath(),
-      );
-
-      logger.tipCommand(
-        "add a foundation to configure your cloud platforms",
-        "foundation new",
-      );
-    });
+    .command("init [directory]")
+    .description("initialize a new collie repository in a new directory")
+    .action(
+      async (opts: GlobalCommandOptions, directoryArg: string | undefined) => {
+        const kit = new CollieRepository("./");
+        const logger = new Logger(kit, opts);
+
+        const dir = new DirectoryGenerator(WriteMode.skip, logger);
+
+        const directory: string = directoryArg ||
+          await Input.prompt(
+            `Choose a directory name for your new collie repository`,
+          );
+        const repo: Dir = {
+          name: directory,
+          entries: [],
+        };
+
+        await dir.write(repo, "");
+        Deno.chdir(directory);
```

Comment:
d: that's kind of dangerous as you're modifying global state (and that's quite hidden as well). That changes the processes CWD. 

I know there's a few open bugs (waiting to happen) in collie-cli that don't do this properly, but we should strive to always operate on fully resolved paths (hence all the `resolve()` helper calls) to remove any ambiguity. This leaves the entry points (command handlers) into collie to establish CWD and then there's no more CWD dependent code anywhere else.

Since you're all inside a command handler here that's kind of ok, but still it has some non-obvious side effects that require carefully reading the code.

e.g. the `dir.write(repo, "")` on L33 will write to a different dir than the one on L65 because you Deno.chdir'd in between. 

I think this would also silently change the behavior of `CollieRepository.resolvePath()` btw!

I think it would be be better to
- pass in an explicit cwd to `git init` (the process runners support this).
- create the CollieRepository only after you git inited (it doesn't make sense to construct one before the repo exists)

Reply:
Thanks, I adapted git init to take a path